src/userspace/exec_monitor.cpp

```diff
@@ -65,11 +66,31 @@ int ExecMonitor::run()
 		std::cerr << "[EXEC_MONITOR]: Error allocating the ringbufffer.\n";
 		goto out;
 	}
-	
+
+	if (n_proc > 0) {
+		skel->bss->n_monitored_proc = 0;
+		skel->bss->max_n_proc = n_proc;
```

Comment:
Mhh I doubt the necessity of having a counter in kernel space. The only benefit of having a kernel space counter is that you never miss an execution (for example if the ring buffer is full).

However, I imagine that if the user gives hawk a maximum number of execution of 50, what the user wants is 50 execution events outputted by hawk. So the count that the user is actually interested in is the number of calls to process_sample. I feel that this is what should stop the loop. This also makes the PR much simpler

Reply:
Do I have any control over how many times process_sample is executed? I imagined that the poll function can result in more than one record added, something related to multithreading; so are you saying that I could make a for instead of while true and poll for data inside that for?

Reply:
If you suspect that poll can wake up once for multiple events (I'm not sure about it, but it's possible), you could also have a counter in your export_data and bail out once you reached the max count.

Reply:
Yes, I could do this. I was of this initially, but the reason I did not go for this idea is because it could mean that I end up adding more records on the ringbuffer than I intent to process and actually use. This is why I have the global variable shared between userspace and kernelspace units, so I make sure that for the Nth + 1 record, I don't even register it on the ringbuffer. I have no preference on this, I could make it work either way, I just wanted to share this thought process with you.

Reply:
Ok, I see. Having a few events on the ring-buffer is okay though. Since hawk stops after it reached N events, the eBPF programs will be detached and the ring-buffer will be destroyed so it's not a big deal.